client-metrics-reporter/src/test/java/io/strimzi/kafka/metrics/prometheus/integration/TestMirrorMakerMetricsIT.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.metrics.prometheus.integration;
+
+import io.strimzi.kafka.metrics.prometheus.ClientMetricsReporter;
+import io.strimzi.kafka.metrics.prometheus.ClientMetricsReporterConfig;
+import io.strimzi.kafka.metrics.prometheus.MetricsUtils;
+import io.strimzi.kafka.metrics.prometheus.http.Listener;
+import io.strimzi.test.container.StrimziConnectCluster;
+import io.strimzi.test.container.StrimziKafkaCluster;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.utility.MountableFile;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class TestMirrorMakerMetricsIT {
+
+    private static final int PORT = Listener.parseListener(ClientMetricsReporterConfig.LISTENER_CONFIG_DEFAULT).port;
+    private static final String CONNECT_ID = "my-cluster";
+    private static final String TOPIC = "input";
+    private static final String GROUP = "my-group";
+    private static final String SOURCE_CONNECTOR = "source";
+    private static final String CHECKPOINT_CONNECTOR = "checkpoint";
+
+    private StrimziKafkaCluster kafka;
+    private StrimziConnectCluster connect;
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        // Use a single cluster as source and target
```

Comment:
hmmm, when you also used the target Kafka cluster, it didn't work, I assume (why?)

Reply:
To be honest I don't really know. I'm wondering if it may have been a resources issue. It seems out of the 2 pipelines (Java 11 and 17) each time a random one would pass and the other fail. Since we don't effectively need 2 Kafka clusters, I opted to use a single one, that also simplified the test slightly.

Reply:
>  I'm wondering if it may have been a resources issue.

I think that's not a problem for sure. As we have in `strimzi-kafka-repository` tests (specifically in STs), which run much more than this.  

> It seems out of the 2 pipelines (Java 11 and 17) each time a random one would pass and the other fail. Since we don't effectively need 2 Kafka clusters, I opted to use a single one, that also simplified the test slightly.

But locally, it was okay, I assume?

---

I was just curious, if that is okay with you, I am fine with that (to have just one Kafka cluster as source/target).

Reply:
I've run this test 100s of times locally and it was consistently passing.